komodo/matrix.py

```diff
@@ -24,10 +22,12 @@ def format_release(base, rhel_ver, py_ver):
 
 def get_matrix_base(release):
     """Return the base (e.g. matrix part of a concrete release). Should be the
-    inverse of format_release for actual, concrete matrix releases."""
-    for rhel_ver, py_ver in get_matrix():
-        suffix = format_release("", rhel_ver, py_ver)
-        if release.endswith(suffix):
-            return release.split(suffix, 1)[0]
+    inverse of format_release for actual, concrete matrix releases.
+    Hard-coded the suffix pattern '-py..-rhel.' or '-py...-rhel.'"""
+
+    if re.search("-py..-rhel.", release):
```

Comment:
You should be able to use capturing groups (using parentheses) and more specific wildcards, like `[0-9]` or `\d` for digits, to capture everything 'at once' here. Something like `\d{2,3}` means "2 or 3 digits", for example. 

Here's one of those sites you can practise regex on: https://regex101.com/ -- and we can take a look together if you want.

Reply:
Excellent tip!